src/js/libgeo/GeoOps.js

```diff
@@ -584,6 +584,43 @@ geoOps.OtherPointOnCircle.updatePosition = function(el) {
     el.homog = General.withUsage(pos, "Point");
 };
 
+
+// start and end assumed to be normalized at once!
+geoOps._helper.projectPointToSegmentCR = function(seg, pos, infseg) {
+    var line = seg.homog;
+    var tt = List.turnIntoCSList([line.value[0], line.value[1], CSNumber.zero]);
+    var farpoint = List.sub(seg.startpos, seg.endpos);
+    var cr = List.crossratio3(
+        farpoint, seg.startpos, seg.endpos, pos, tt);
+
+    //handle case if we have infinite segment
+    if (infseg) {
```

Comment:
What exactly is an “infinite segment”? If it is a projectively transformed segment which passes through infinity, then the fact that we compute `farpoint` as the projectively transformed point at infinity should already take care of this. I think `examples/62_ProjTransform.html` demonstrates this: it has a yellow point on a projected segment, and does the right thing even if said segment should pass through infinity. If you disagree, do you have an example which demonstrates why this would be needed?